### PR TITLE
Swaps initial UI

### PR DESCRIPTION
**Pull request by christopher-dembski**
### Description

Part of https://github.com/christopher-dembski/delta/issues/30

Initial implementation of the UI for creating swaps.

Currently, the user is presented with a multi-stage form, where they can select a goal type and an intensity if they select an imprecise goal.

**Follow-ups:** 
- Integrating this form with nutrients, so the user is prompted to select a nutrient for their goal. 
- Passing data to backend to generate swaps
- Rendering meal statistics for old vs. new meal

### Testing

**Video**

**Note:** The video has to bee small for GitHub, so I don't have time to demonstrate all edge cases. I encourage reviewers to test manually in their own environment. 

https://github.com/user-attachments/assets/28cc097d-67df-4832-aad4-933220f67ea3

**Integration Tests**

I opted for integration test between the view and the UI. I started by writing the tests for the presenter using Mockito to mock the view. But, the mocks became really extensive, where they were time-consuming to write and almost nothing was actually being tested! When the backend swaps services are introduced, this test will get more extensive. The plan is to mock the backend services and verify that they are called with the correct input after a series of interactions with the UI (for example, called with an "imprecise" goal with "high" intensity) if the user selects these items.

### CODE REVIEW

File: src/main/java/swaps/view/ChooseOneOrTwoGoalsCard.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    public void setOneOrTwoGoalsDropdown(DropDownCreateSecondGoal createSecondGoal) {
+        oneOrTwoGoalsDropdown.setSelectedItem(createSecondGoal);
+    }
+
+    public void addOneOrTwoGoalsDropdownListener(Consumer<DropDownCreateSecondGoal> listener) {
+        oneOrTwoGoalsDropdown.addActionListener(e -> {
+            DropDownCreateSecondGoal yesNo = (DropDownCreateSecondGoal) oneOrTwoGoalsDropdown.getSelectedItem();
+            listener.accept(yesNo);
+        });
+    }
```
Comment (christopher-dembski) [inline src/main/java/swaps/view/ChooseOneOrTwoGoalsCard.java:28]: This same pattern is used throughout. Essentially, when the dropdown is fired, call the provide function `listener` with the selected item from the dropdown.

File: src/main/java/swaps/view/ChooseOneOrTwoGoalsCard.java
Comment (christopher-dembski) [inline src/main/java/swaps/view/ChooseOneOrTwoGoalsCard.java:1]: I separated some components into smaller classes to avoid one large class with everything. Each one of these cards is shown in sequence, and the user navigates between them with  "Previous" and "Next" buttons.

File: src/main/java/swaps/view/DropDownCreateSecondGoal.java
Comment (christopher-dembski) [inline src/main/java/swaps/view/DropDownCreateSecondGoal.java:1]: I might refactor to create `DropDown` super class to remove some duplicated logic. Basically, this defines an enum representing options in a dropdown (combo-box) that can be represented as a string in the dropdown.

File: src/main/java/swaps/view/ICreateImpreciseGoalCard.java
Comment (christopher-dembski) [inline src/main/java/swaps/view/ICreateImpreciseGoalCard.java:1]: I'm not sure these interfaces really add much value. Thoughts on whether we should keep them? Its quite a bit of boiler -plate and I don't really see the advantage in this circumstance.

File: src/main/java/swaps/view/SwapsPresenter.java
Comment (christopher-dembski) [inline src/main/java/swaps/view/SwapsPresenter.java:1]: The presenter contains all the UI logic and the application state. It contains no Java Swing UI components.

File: src/test/java/swaps/TestSwapsUI.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+import static org.mockito.Mockito.*;
+
+class TestSwapsUI {
+    private SwapsPresenter presenter;
+    private ISwapsView view;
+
+    @BeforeEach
+    public void beforeEach() {
+        view = spy(new SwapsView());
```
Comment (christopher-dembski) [inline src/test/java/swaps/TestSwapsUI.java:18]: Spying on an object creates a proxy that lets you determine whether certain methods were called and with what values. Think of it as intercepting the method calling, logging that the method was called in with what parameters, and then calling the actual method.

**Comment by christopher-dembski** [comment]
Closing in favor of a simpler appraoch
